qiskit_machine_learning/kernels/algorithms/quantum_kernel_trainer.py

```diff
@@ -198,13 +197,17 @@ def fit(
             msg = "Quantum kernel cannot be fit because there are no user parameters specified."
             raise ValueError(msg)
 
-        # Bind inputs to objective function
-        output_kernel = copy.deepcopy(self._quantum_kernel)
-
         # Randomly initialize the initial point if one was not passed
         if self._initial_point is None:
             self._initial_point = algorithm_globals.random.random(num_params)
 
+        # Bind inputs to objective function
+        output_kernel = type(self._quantum_kernel)(
+            feature_map=self._quantum_kernel.feature_map,
+            training_parameters=self._quantum_kernel.training_parameters,
+        )
```

Comment:
I'm not sure this is enough. For instance, `TrainableFidelityQuantumKernel` has `fidelity` and here is it omitted. So, the `output_kernel` will pick the default implementation of fidelity which is based on the reference primitives, I guess. Same with other parameters.

Reply:
That is a good point, it requires investigation by @oscar-wallis . However,  nobody can use this algorithm for the hardware with the current status, we need a quick fix for the deep copy problem. One simple temporary solution is directly using the given TrainableFidelityQuantumKernel, but it will change the input (as a user I generally dont mind).

Reply:
As a quick fix it is fine. But then, it should be written somewhere that this is a temp solution with certain limitations. Although this fix won't help running QKT on hardware as once `output_kernel` is created it will default to a simulator instead of hardware.